Add a block attributes filter

lukecarbis opened the pull request:
cc. @phpbits

phpbits committed:
Add blocklab_get_block_attributes for https://github.com/getblocklab/block-lab/issues/360

lukecarbis committed:
Merge pull request #361 from phpbits/add/blocklab_get_block_attributes_filter

Add `blocklab_get_block_attributes` filter

lukecarbis committed:
Add underscore to filter name

lukecarbis commented:
@kienstra This is a real quick win, if you could please review?

kienstra commented on php/blocks/class-loader.php:
```diff
@@ -187,7 +187,7 @@ public function get_block_attributes( $block ) {
 		}
 
-		return $attributes;
+		return apply_filters( 'block_lab_get_block_attributes', $attributes );
```
@lukecarbis, would you mind if I committed a DocBlock for `apply_filters` here, including a `@param` tag for `$attributes`?

lukecarbis replied:
Of course! I should have thought of that. Go for it.

kienstra replied:
Thanks!

kienstra committed:
Add a DocBlock @param tag for $attributes

@todo: consider passing the $block to the filter,
as this doesn't have much context.

kienstra committed:
Also pass the $block to the filter, especially to get the $block['name']

Before, there was no way to get which block this was.
Maybe this isn't an important use-case, though.

kienstra commented:
**Passing `$block` to filter?**

Hi @lukecarbis, 
what do you think about https://github.com/getblocklab/block-lab/pull/363/commits/3c335e49d477a9bd5e55f163a40a09c8cd3b5217, which also passes `$block` to the filter?

Otherwise, it might be hard to know which block it's for:

<img width="1013" alt="attributes-value" src="https://user-images.githubusercontent.com/4063887/61016776-76f82680-a356-11e9-8608-60b37136fcfd.png">

kienstra commented:
Also, I'm working on the failed Travis build in #364.

Commit:
Merge branch 'develop' into add/get-block-attributes-filter

lukecarbis commented:
@kienstra Thanks for those two commits. They both look great! Especially 3c335e49d477a9bd5e55f163a40a09c8cd3b5217 – good thinking. Ready for you to re-review.

kienstra reviewed (approved):
**Approved**

Thanks, @lukecarbis! This should be ready to merge.
